lightning-invoice/src/payment.rs

```diff
@@ -1585,7 +1945,7 @@ mod tests {
 				return;
 			}
 
-			if let Some(expectations) = &self.expectations {
+			if let Some(expectations) = &self.event_expectations {
 				if !expectations.is_empty() {
 					panic!("Unsatisfied scorer expectations: {:?}", expectations);
 				}
```

Comment:
Could you do this for both sets of expectations? Otherwise, an unexercised expectation won't cause a test failure.